trace/tests/unit/test__gax.py

```diff
@@ -0,0 +1,377 @@
+        END_TIME = datetime.utcnow()
+
+        start_time_pb = _datetime_to_pb_timestamp(START_TIME)
+        end_time_pb = _datetime_to_pb_timestamp(END_TIME)
+
+        span_pb = TraceSpan(
+            span_id=SPAN_ID,
+            name=SPAN_NAME,
+            start_time=start_time_pb,
+            end_time=end_time_pb)
+
+        trace_pb = Trace(
+            project_id=PROJECT,
+            trace_id=TRACE_ID,
+            spans=[span_pb])
+
+        parse_result = self._call_fut(trace_pb)
+
+        expected_result = {
+            'projectId': PROJECT,
+            'traceId': TRACE_ID,
+            'spans': [
+                {
+                    'spanId': str(SPAN_ID),
+                    'name': SPAN_NAME,
+                    'startTime': START_TIME.isoformat() + 'Z',
+                    'endTime': END_TIME.isoformat() + 'Z',
+                },
+            ],
+        }
+
+        self.assertEqual(parse_result, expected_result)
+
+    @mock.patch('google.cloud.trace._gax.MessageToDict',
+                side_effect=TypeError)
+    def test_unregistered_type(self, msg_to_dict_mock):
+        trace_pb = mock.Mock(spec=['HasField'])
+        trace_pb.HasField.return_value = False
+        with self.assertRaises(TypeError):
+            self._call_fut(trace_pb)
+
+
+class Test_make_gax_trace_api(unittest.TestCase):
+
+    def _call_fut(self, client):
+        from google.cloud.trace._gax import make_gax_trace_api
+
+        return make_gax_trace_api(client)
+
+    def test_it(self):
+        from google.cloud.trace._gax import _TraceAPI
+        from google.cloud._http import DEFAULT_USER_AGENT
+
+        credentials = object()
+        client = mock.Mock(_credentials=credentials, spec=['_credentials'])
+        channels = []
+        channel_args = []
+        generated_api_kwargs = []
+        channel_obj = object()
+        generated = object()
+
+        def make_channel(*args):
+            channel_args.append(args)
+            return channel_obj
+
+        def generated_api(channel=None, **kwargs):
+            channels.append(channel)
+            generated_api_kwargs.append(kwargs)
+            return generated
+
+        host = 'foo.apis.invalid'
+        generated_api.SERVICE_ADDRESS = host
+
+        patch = mock.patch.multiple(
+            'google.cloud.trace._gax',
+            TraceServiceClient=generated_api,
+            make_secure_channel=make_channel)
+        with patch:
+            trace_api = self._call_fut(client)
+
+        self.assertEqual(channels, [channel_obj])
+        self.assertEqual(channel_args,
+                         [(credentials, DEFAULT_USER_AGENT, host)])
+
+        self.assertEqual(len(generated_api_kwargs), 1)
+        self.assertEqual(generated_api_kwargs[0]['lib_name'], 'gccl')
+
+        self.assertIsInstance(trace_api, _TraceAPI)
+        self.assertIs(trace_api._gax_api, generated)
+        self.assertIs(trace_api.client, client)
+
+
+class _GAXTraceAPI(_GAXBaseAPI):
```

Comment:
You can use a mock, maybe @dhermes can share some specific recommendations for this case.